jans-cedarling/bindings/cedarling_c/tests/test_cedarling.c

```
@@ -0,0 +1,499 @@
+    if(ret==0){
+        TEST_ASSERT(auth_result.DATA!=NULL,"Authorization result data provided");
+        
+        // Cast uint32_t* to char* for string operations
+        char* result_str = (char*)auth_result.DATA;
+        printf(" Authorization result: %.200s...\n", result_str);
+        
+        // Verify the response structure contains expected fields
+        // Note: The overall decision depends on principal_bool_operator config (not exposed to C)
+        // Default operator requires Jans::Workload AND Jans::User, so decision will be false
+        // when using TestPrincipal1. We verify the individual principal got "allow".
+        TEST_ASSERT(strstr(result_str, "\"decision\":\"allow\"") != NULL,
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)